tests/spread/core22/project-variables/snapcraft.yaml

```diff
@@ -0,0 +1,57 @@
+name: project-variables
+base: core22
+version: "1.0.0"
+summary: Test evaluation of project variables in keys.
+
+# Description is a convenient key for testing project variables because it is freeform.
+# Note that this test doesn't test project variables in override scriplets,
+# since those variables are set in the build environment and are tested separately.
+description: |
+  CRAFT_ARCH_BUILD_FOR           = $CRAFT_ARCH_BUILD_FOR
+  CRAFT_ARCH_BUILD_FOR           = ${CRAFT_ARCH_BUILD_FOR}
```

Comment:
This is not obvious to me why we test both env var syntax for all variables. Is that because we expect that one of them could yield a different result while others keep working fine?

Reply:
Exactly.  If these were expanded by a shell, I wouldn't test this.

However, these variables are expanded by [craft-parts](https://github.com/canonical/craft-parts/blob/0681571f2c451709977a4c56a302625d39ab4bcd/craft_parts/executor/environment.py#L306-L309) and we've had a history of variables breaking differently, depending if they have curly braces or not ([1](https://github.com/canonical/craft-parts/pull/554), [2](https://github.com/canonical/snapcraft/issues/5500))